builder/vsphere/common/cleanup_vm.go

```diff
@@ -0,0 +1,30 @@
+package common
+
+import (
+	"github.com/hashicorp/packer/builder/vsphere/driver"
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+)
+
+func CleanupVM(state multistep.StateBag) {
+	_, cancelled := state.GetOk(multistep.StateCancelled)
+	_, halted := state.GetOk(multistep.StateHalted)
+	_, destroy := state.GetOk("destroy_vm")
```

Comment:
This checks that destroy_vm is in state but it doesn't check its value. Is that what you meant?

Reply:
This would have the same outcome:

https://github.com/hashicorp/packer/blob/942bfbf2214b2d2cdc42002c23c08571ad2a2d74/builder/vsphere/common/step_import_to_content_library.go#L137-L142